libtenzir/include/tenzir/format/syslog.hpp

```diff
@@ -250,6 +254,131 @@ struct message_parser : parser_base<message_parser> {
+    const auto word = +(parsers::printable - space);
+    const auto ws = +space;
+    const auto wsignore = ignore(ws);
+    const auto is_month = [](const std::string& mon) {
+      return mon == "Jan" || mon == "Feb" || mon == "Mar" || mon == "Apr"
+             || mon == "May" || mon == "Jun" || mon == "Jul" || mon == "Aug"
+             || mon == "Sep" || mon == "Oct" || mon == "Nov" || mon == "Dec";
+    };
+    const auto is_day = [&](const std::string& day) -> bool {
+      const auto p = integral_parser<uint16_t, 2, 1>{}.with([](uint16_t day) {
+        return day <= 31;
+      });
+      auto sv = std::string_view{day};
+      const auto* f = sv.begin();
+      const auto* const l = sv.end();
+      return p(f, l, unused) && f == l;
+    };
+    const auto is_time = [&](const std::string& time) -> bool {
+      const auto hour_parser
+        = integral_parser<uint16_t, 2, 2>{}.with([](uint16_t hour) {
+            return hour <= 23;
+          });
+      const auto minsec_parser
+        = integral_parser<uint16_t, 2, 2>{}.with([](uint16_t x) {
+            return x <= 59;
+          });
+      const auto p
+        = hour_parser >> ':' >> minsec_parser >> ':' >> minsec_parser;
+      auto sv = std::string_view{time};
+      const auto* f = sv.begin();
+      const auto* const l = sv.end();
+      return p(f, l, unused) && f == l;
+    };
+    const auto timestamp_parser
+      = (word.with(is_month) >> ws >> word.with(is_day) >> ws
+         >> word.with(is_time))
+          ->*[](std::tuple<std::string, std::string, std::string, std::string,
+                           std::string>
+                  x) {
+                return fmt::to_string(fmt::join(x, ""));
+              };
+    const auto tag_parser = +(parsers::printable - ':');
+    const auto content_parser = ~(wsignore >> *parsers::printable);
+    // clang-format off
+    const auto p = ~(priority_parser >> wsignore)
+                   >> timestamp_parser >> wsignore    // timestamp
+                   >> word >> wsignore                // host
+                   >> tag_parser >> ':'               // tag
+                   >> content_parser;                 // content
+    // clang-format on
+    if constexpr (std::is_same_v<Attribute, unused_type>)
+      return p(f, l, unused);
+    else
+      return p(f, l, x.timestamp, x.host, x.tag, x.content);
+  }
+};
+
+inline type make_syslog_type() {
+  return type{
+    "syslog.rfc5424",
+    record_type{{
+      {"facility", uint64_type{}},
+      {"severity", uint64_type{}},
+      {"version", uint64_type{}},
+      {"timestamp", time_type{}},
+      {"hostname", string_type{}},
+      {"app_name", string_type{}},
+      {"process_id", string_type{}},
+      {"message_id", string_type{}},
+      {"structured_data",
+       map_type{type{"id", string_type{}},
+                type{"params", map_type{type{"key", string_type{}},
+                                        type{"value", string_type{}}}}}},
```

Comment:
You can leave the re-introduction of `structure_data` to a follow-up PR, in particular because the `map` type should not be used anymore, see https://github.com/tenzir/issues/issues/25. So this either requires upgrading to the new `series_builder` or using a `{key: ..., value: ...}` workaround.